src/ctia/http/server_service_core.clj

```diff
@@ -1,18 +1,57 @@
 (ns ctia.http.server-service-core
-  (:require [ctia.http.server :refer [new-jetty-instance]]
-            [ctia.schemas.core :refer [APIHandlerServices]]
+  (:require [ctia.graphql.schemas :as graphql.schemas]
+            [ctia.http.server :refer [new-jetty-instance]]
+            [ctia.schemas.core :refer [APIHandlerServices
+                                       APIHandlerServices->RealizeFnServices
+                                       resolve-with-rt-ctx]]
             [clojure.tools.logging :as log]
-            [schema.core :as s])
-  (:import [org.eclipse.jetty.server Server]))
+            [schema.core :as s]
+            [schema-tools.core :as st])
+  (:import [graphql GraphQL]
+           [org.eclipse.jetty.server Server]))
+
+(defn get-graphql [{:keys [graphql]}]
+  {:post [(instance? GraphQL %)]}
+  graphql)
+
+(s/defn ^:private server->port :- (s/constrained s/Int pos?)
+  [server :- Server]
+  (-> server
+      .getURI
+      .getPort))
 
 (s/defn start [context
-               {:keys [port] :as http-config}
-               services :- APIHandlerServices]
-  (log/info (str "Starting HTTP server on port " port))
-  (assoc context
-         :services services
-         :server (new-jetty-instance http-config services)))
+               http-config
+               services :- (-> APIHandlerServices
+                               (st/dissoc-in [:CTIAHTTPServerService :get-port])
+                               (st/dissoc-in [:CTIAHTTPServerService :get-graphql]))]
```

Comment:
since `CTIAHTTPServerService` only contains what is dissoced, I would simply dissoc `:CTIAHTTPServerService` or optionalize it's keys.